### ISSUE TITLE
System.Collections.Generic.Comparer<T> Default and Create issue in sorting

### ISSUE THREAD

**Issue by d-p-y** [bridgedotnet/Bridge#2174]
### Expected

I expect that ```System.Collections.Generic.Comparer<T>.Default``` and ```System.Collections.Generic.Comparer<T>.Create(IComparison<T>)``` can be called generically. Specifically it means they have generic (alias) method named ```System$Collections$Generic$IComparer$1$T$compare```

### Actual

Actually they don't have it as evidenced by code that calls ```IComparer<T>.Compare(T x, T y)``` in a generic way. 

*Without too much abstraction*: it means that currently it is impossible to sort using code that delegates its work to ```Comparer<T>.Default``` 

### Steps To Reproduce

[Deck](http://deck.net/7fab190beb9d1dc1420e59256a835dc9)

Test below should write 6 messages ending with true. In Bridge 15.6 it throws an exception while producing 5th and 6th line. I suspect it can be broken in older versions. 

```cs
public class CustomCmp<T> : IComparer<T> where T : IComparable<T>
{
    public int Compare(T x, T y)
    {
        return x.CompareTo(y);
    }
}

public class WrappingCmp<T> : IComparer<T>
{
    private readonly IComparer<T> _wrapped;

    public WrappingCmp(IComparer<T> wrapped)
    {
        _wrapped = wrapped;
    }

    public int Compare(T x, T y)
    {
        return _wrapped.Compare(x, y);
    }
}
    
public class Main
{
    public static void Main()
    {
        //comparer Default "as such" works
        {
            var cmp = Comparer<string>.Default;
            Console.WriteLine("[1]is less than zero as expected?" + (cmp.Compare("a", "b") < 0));
        }
        
        //comparer Create "as such" works
        {
            var cmp = Comparer<string>.Create((x,y) => x.CompareTo(y));

            Console.WriteLine("[2]is less than zero as expected?" + (cmp.Compare("a", "b") < 0));
        }

        //custom comparer "as such" works
        {
            var cmp = new CustomCmp<string>();
            Console.WriteLine("[3]is less than zero as expected?" + (cmp.Compare("a", "b") < 0));
        }
        
        //custom comparer wrapped works
        {
            var cmp = new WrappingCmp<string>(new CustomCmp<string>());
            Console.WriteLine("[4]is less than zero as expected?" + (cmp.Compare("a", "b") < 0));
        }
        
        //default comparer wrapped doesn't work
        {
            var cmp = new WrappingCmp<string>(Comparer<string>.Default);

            Console.WriteLine("[5]is less than zero as expected?" + (cmp.Compare("a", "b") < 0));
        }
        
        //created comparer wrapped doesn't work
        {
            var cmp = new WrappingCmp<string>(Comparer<string>.Create((x,y) => x.CompareTo(y)));

            Console.WriteLine("[6]is less than zero as expected?" + (cmp.Compare("a", "b") < 0));
        }
    }
}
```

**Comment by LeonidVeriga** [comment]
Hi @d-p-y, thank you for the great report and test case. It is fixed and will be released in **15.7**.

**Comment by d-p-y** [comment]
thanks!
